datamol/reactions/_reactions.py

```diff
@@ -160,7 +160,8 @@ def is_reaction_ok(rxn: dm.ChemicalReaction, enable_logs: bool = False) -> bool:
 
 
 def select_reaction_output(
-    product: list,
+    product: Union[list, tuple],
+    product_index: int = None,
```

Comment:
Should be `product_index: Optional[int] = None`

Reply:
Also please add in the docstring what it means when it's None.

Reply:
I tihnk `Union[list, tuple]` can be replaced by `Sequence` (`from typing import Sequence`).

Reply:
For `product` . Can you also specify the type within the sequence ?

Reply:
It sees so.  So for product, it's `Sequence[Sequence[dm.mol]]`.